app/script/entity/Conversation.js

```diff
@@ -335,16 +335,18 @@ z.entity.Conversation = class Conversation {
   /**
    * Adds a single message to the conversation.
    * @param {z.entity.Message} messageEntity - Message entity to be added to the conversation.
-   * @param {boolean} replaceDuplicate - If a duplicate already exists, replace it with the new entity.
+   * @param {boolean} replaceDuplicate - If a duplicate (or a message that should be replaced) already exists, replace it with the new entity.
    * @returns {undefined} No return value.
    */
   add_message(messageEntity, replaceDuplicate = false) {
     if (messageEntity) {
-      const duplicateEntity = this._findDuplicate(messageEntity);
+      const entityToReplace =
+        this._findDuplicate(messageEntity.id, messageEntity.from) ||
+        this._findDuplicate(messageEntity.replacing_message_id, messageEntity.from);
```

Comment:
Let's move these into separate const that detail what kind of duplicate we retrieved.